src/coordinator/functions/logical/or.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logical
+
+import (
+	"github.com/m3db/m3db/src/coordinator/block"
+)
+
+func makeOrBuilder(
+	logicalNode *BaseNode,
+	lIter, rIter block.StepIter,
+) (block.Block, error) {
+	meta, err := combineMetadata(lIter.Meta(), rIter.Meta())
+	if err != nil {
+		return nil, err
+	}
+	missingIndices, combinedSeriesMeta := missing(
+		logicalNode.op.Matching,
+		lIter.SeriesMeta(),
+		rIter.SeriesMeta(),
+	)
+
+	builder, err := logicalNode.controller.BlockBuilder(meta, combinedSeriesMeta)
+	if err != nil {
+		return nil, err
+	}
+	if err := builder.AddCols(lIter.StepCount()); err != nil {
+		return nil, err
+	}
+
+	index := 0
+	for ; lIter.Next(); index++ {
+		lStep, err := lIter.Current()
+		if err != nil {
+			return nil, err
+		}
+		lValues := lStep.Values()
+		builder.AppendValues(index, lValues)
+	}
+
+	if err := addValuesAtIndeces(missingIndices, rIter, builder); err != nil {
+		return nil, err
+	}
+
+	return builder.Build(), nil
+}
+
+// missing returns the slice of rhs indices for which there are no corresponding
```

Comment:
update comments ? Since now missing is returning the combined series meta as well